futures-core/src/impls.rs

```diff
@@ -0,0 +1,132 @@
+use {task, Future, Stream, Poll, Unpin};
+
+#[cfg(feature = "either")]
+use either::Either;
+
+unsafe impl<T: Unpin> Unpin for Box<T> {}
+unsafe impl<T: Unpin> Unpin for Option<T> {}
+unsafe impl<'a, T: Unpin> Unpin for &'a T {}
+unsafe impl<'a, T: Unpin> Unpin for &'a mut T {}
```

Comment:
Also `unsafe impl<T: Unpin> Unpin for AssertUnwindSafe<T> {}`

Reply:
`Box<T>` should be unconditionally `Unpin` (according to the current understanding in the `Pin` RFC issue).

Reply:
What? Why? Then we couldn't implement `Future` for `Box<T: Future + !Unpin>`